x-pack/plugins/observability_solution/apm/server/routes/traces/get_aggregated_critical_path.ts

```diff
@@ -123,38 +123,38 @@ export async function getAggregatedCriticalPath({
                 double duration;
                 
                 def operationMetadata = [
-                  "service.name": doc['service.name'].value,
-                  "processor.event": doc['processor.event'].value,
-                  "agent.name": doc['agent.name'].value
+                  "service.name": $('service.name', ''),
+                  "processor.event": $('processor.event', ''),
+                  "agent.name": $('agent.name', '')
                 ];
 
-                def isSpan = !doc['span.id'].empty && !doc['span.name'].empty;
-                
-                if (isSpan) {
-                  id = doc['span.id'].value;
-                  operationMetadata.put('span.name', doc['span.name'].value);
-                  if (!doc['span.type'].empty) {
-                    operationMetadata.put('span.type', doc['span.type'].value);
+                id = $('span.id', null);
+                if (id != null) {
+                  operationMetadata.put('span.name', $('span.name', ''));
+                  def spanType = $('span.type', '');
+                  if (spanType != '') {
+                    operationMetadata.put('span.type', spanType);
                   }
-                  if (!doc['span.subtype'].empty) {
-                    operationMetadata.put('span.subtype', doc['span.subtype'].value);
+                  def spanSubtype = $('span.subtype', '');
+                  if (spanSubtype != '') {
+                    operationMetadata.put('span.subtype', spanSubtype);
                   }
-                  duration = doc['span.duration.us'].value;
+                  duration = $('span.duration.us', 0);
                 } else {
-                  id = doc['transaction.id'].value;
-                  operationMetadata.put('transaction.name', doc['transaction.name'].value);
-                  operationMetadata.put('transaction.type', doc['transaction.type'].value);
-                  duration = doc['transaction.duration.us'].value;
+                  id = $('transaction.id', '');
+                  operationMetadata.put('transaction.name', $('transaction.name', ''));
+                  operationMetadata.put('transaction.type', $('transaction.type', ''));
+                  duration = $('transaction.duration.us', 0);
                 }
                  
                 String operationId = toHash(operationMetadata);
                 
                 def map = [
-                  "traceId": doc['trace.id'].value,
+                  "traceId": $('trace.id', ''),
                   "id": id,
-                  "parentId": doc['parent.id'].empty ? null : doc['parent.id'].value,
+                  "parentId": $('parent.id', null),
                   "operationId": operationId,
-                  "timestamp": doc['timestamp.us'].value,
+                  "timestamp": $('timestamp.us', 0),
```

Comment:
what happens if all these are null/empty? do we filter them out or forward return them in the result? are there any implications if they end up in the frontend?

Reply:
`trace.id` and `timestamp.us` will always be set for transactions and spans. `parent.id` may be missing, but it would previously have been added to the map as `null` in that case too. Anyway, these attributes don't actually show up in the frontend - they are used only during the aggregation.